pym/bob/archive.py

```diff
@@ -885,6 +885,13 @@ def __retry(self, request):
                 if retries == 0: return (False, e)
                 retries -= 1
 
+    def _retry(self, request):
+        (ok, result) = self.__retry(request)
+        if ok:
+            return result
+        else:
+            raise result
+
```

Comment:
Why not just modify `__retry` accordingly? AFAICS, all call sites rely on throwing in case of an error.

Also, please separate the cleanup (getting rid of all the identical if/else constructs) from the actual change. Those are two independent changes.